### ISSUE TITLE
[QUESTION]: hyperparameter optimization in Python 

### ISSUE THREAD

**Issue by iwwwish** [chemprop/chemprop#424]
**What are you trying to do?**
I would like to use optimal hyperparameters before I train a model in my pipeline. I could do cross validation and prediction directly in my Python code. But I am unable to figure out how to use `hyperparameter_optimization.py` directly within Python. I'd imagine it to be similar to cross validation and tried it the following way.

**Previous attempts**
```
# hyperopt arguments
self.hyperopt_args = [
    '--data_path', self.train_data_path,
    '--dataset_type', 'regression',
    '--num_iters', '10',
    '--config_save_path', self.hyperopt_config_path,
]

# perform hyperopt
args_hyperopt = chemprop.args.HyperoptArgs().parse_args(self.hyperopt_args)
chemprop.hyperparameter_optimization.hyperopt(args=args_hyperopt) # does not return anything
```

**Output**
After running this, the log prints the following and nothing happens after:

```
Creating search space using parameters ['dropout', 'linked_hidden_size', 'ffn_num_layers', 'depth'].
```

I even tried this on a GPU assuming hyperopt might need a more compute power. Any idea of how to do this? Appreciate your help. Thank you!

Vishal

PS: Taking the command line entry to hyperparameter optimization works as expected.

**Comment by kevingreenman** [comment]
Hi Vishal, thanks for your question. I've copied your code into the [Colab demo notebook](https://github.com/chemprop/chemprop/blob/master/colab_demo.ipynb) for testing and am unable to reproduce your issue using the test regression dataset. When I run the following, hyperparameter optimization runs as expected.
```
hyperopt_args = [
    '--data_path', 'data/regression.csv',
    '--dataset_type', 'regression',
    '--num_iters', '10',
    '--config_save_path', '.',
]

args_hyperopt = chemprop.args.HyperoptArgs().parse_args(hyperopt_args)
chemprop.hyperparameter_optimization.hyperopt(args=args_hyperopt)
```

Can you think of any reason why the way you're running the script would give different results from the Colab notebook?

**Comment by iwwwish** [comment]
Hi Kevin! Thank you for looking into this. I tried running the same in a Jupyter notebook. Works fine, albeit at a very slow pace. However, I cannot see anything in the logs beyond the following lines when I run the same code as a step in my ML pipeline.

```
2023-07-24 23:19:01.608 [1690265903322432/chemprop_hyperopt/2 (pid 31322)] No manual trials loaded as part of hyperparameter search
2023-07-24 23:19:01.608 [1690265903322432/chemprop_hyperopt/2 (pid 31322)] Initiating trial with seed 0
2023-07-24 23:19:01.608 [1690265903322432/chemprop_hyperopt/2 (pid 31322)] Loaded 0 previous trials
2023-07-24 23:19:01.609 [1690265903322432/chemprop_hyperopt/2 (pid 31322)] Parameters assigned with random search, 50 random trials remaining
```

I confirmed that nothing was going on by looking at CPU/GPU utilization. Not sure why this happens only when I run this as a part of my pipeline. :/

**Comment by kevingreenman** [comment]
That's very odd. After those lines print to the log, is the process with pid 31322 still running (but not using any CPU/GPU), or has the process terminated? You can check this with `ps aux | grep 31322` on the command line. Or just run `ps aux` and look through to see if any other processes related to `chemprop_hyperopt` are still running.

**Comment by iwwwish** [comment]
I did look into that. This time I tried with a tiny dataset of 100 molecules. I figured the the reason it does not print detailed logs could be because the logger in `hyperparameter_optimization.py` script uses the argument `quiet=True` which is otherwise user defined (or) by default set to `False` in case of training. But when I run the same in the Jupiter notebook, it prints the logs as usual. So, not sure what the difference is here. And coming back to the memory utilization, it did drop to 0 after a while but the process was not killed.
